File: Python/Arrays/duplicateinarray.py
def findDuplicate(arr):
    # Write your code here
    n = len(arr)
    for i in range(0, n):
        for j in range(i+1, n):
            if arr[i] == arr[j]:
                return arr[i]
    return -1
    pass


def findDuplicate(arr):
    # Write your code here
    n = len(arr)
    list1 = [0]*n
    for i in range(0, n):
        list1[arr[i]] += 1

    for i in range(0, n):
        if list1[i] > 1:
            return i
    return -1
    pass


def findDuplicate(arr):
    n = len(arr)
    ans = 0
    for i in range(0, n):
        ans = ans ^ arr[i]

    for i in range(0, n):
        ans = ans ^ i
    return ans
    pass


# Example test case
arr = [1, 2, 3, 4, 4]
result = findDuplicate(arr)
if result != -1:
    print(f"Duplicate number: {result}")
else:
    print("No duplicate found.")

File: C++/Arrays/duplicateinarray.cpp
#include <iostream>
#include <climits>
#include <vector>
using namespace std;
// Approach 1: Nested Loops
int findDuplicate(vector<int> &arr)
{
    int n = arr.size();
    for (int i = 0; i < n; i++)
    {
        for (int j = i + 1; j < n; j++)
        {
            if (arr[i] == arr[j])
            {
                // Found a duplicate number
                return arr[i];
            }
        }
    }
    // No duplicate found
    return -1;
}

// Approach 2: Counting using Array
int findDuplicate(vector<int> &arr)
{
    int n = arr.size();
    vector<int> cnt(n, 0); // Array to count occurrences
    for (int i = 0; i < n; i++)
    {
        cnt[arr[i]]++; // Count occurrences of each number
    }
    for (int i = 0; i < n; i++)
    {
        if (cnt[i] > 1)
        {
            // Found a duplicate number
            return i;
        }
    }
    // No duplicate found
    return -1;
}

// Approach 3: Bitwise XOR
int findDuplicate(vector<int> &arr)
{
    int ans = 0;
    for (int i = 0; i < arr.size(); i++)
    {
        ans = ans ^ arr[i]; // Calculate XOR of all elements
    }
    for (int i = 0; i < arr.size(); i++)
    {
        ans = ans ^ i; // XOR with indices
    }
    // Result is the XOR of the duplicate number and indices
    return ans;
}
int main()
{
    // Example test case
    vector<int> arr = {1, 2, 3, 4, 4};
    int result = findDuplicate(arr);
    cout << "Duplicate number: " << result << endl;

    return 0;
}